core/src/data_sources/data_source.rs

```diff
@@ -659,16 +659,49 @@ impl DataSource {
         field_name: &str,
```

Comment:
Better to review with hide whitespace (click [here](https://github.com/dust-tt/dust/pull/5359/files?diff=split&w=1)).